src/Bindings/PluginLua.cpp

```diff
@@ -72,12 +72,13 @@ void cPluginLua::Close(void)
 	cResettablePtrs resettables;
 	std::swap(m_Resettables, resettables);
 	{
-		cCSUnlock Unlock(Lock);
+		m_CriticalSection.Unlock();
 		for (auto resettable: resettables)
 		{
 			resettable->Reset();
 		}
 		m_Resettables.clear();
+		m_CriticalSection.Lock();
```

Comment:
I'd like to try this again. There are two places in the server that use cCSUnlock around something other than a cEvent::Wait(). Would you mind if I had try catch blocks in three places to make this exception safe, and annotatible?